### DISCUSSION TITLE
Algum PSP totalmente no ar?

### DISCUSSION THREAD

**Discussion by flaviolenz** [bacen/pix-api#198] [Q&A]
Alguma IP funcionando com tudo, pelo menos para o QR Estatico?
   (API + webhook + etc.)

**Comment by rubenskuhl** [comment]
Acho que com cobv e lotecobv nenhum, mas cob, pix e webhook, o que cito como confirmado em #76, a Gerencianet. Se alguém souber de mais algum (com API Pix padrão) por favor comente que eu atualizo, tenho buscado mais detalhes com os que alegaram que teriam já no Go Live mas não tenho tido muitas respostas.

Sobre a GN, o que não sei é se eles já estão mandando estático com TxID para o webhook, pois estamos focando no dinâmico. 

Notar que tanto em estático quanto dinâmico há às vezes problemas com o PSP pagador, vide #188 e #192. Há de fato mais problemas com dinâmico e usar estático no momento pode ser sim uma boa mitigação para esses problemas de interoperabilidade.

**Comment by rodrigonunes100** [comment]
Eu achei isso no site do Bradesco, mas não sei se pode ajudar. https://banco.bradesco/pix/assets/docs/api_pix_200.pdf

**Comment by rubenskuhl** [comment]
> Eu achei isso no site do Bradesco, mas não sei se pode ajudar. https://banco.bradesco/pix/assets/docs/api_pix_200.pdf

O documento já detalha pq em #76 Bradesco está citado como usando API divergente, apesar de bem próxima, da API padrão do BACEN. Esses eu não tinha preocupado em colocar a situação operacional, mas de fato o Bradesco consta como operacional segundo quem testou.

**Comment by rodrigonunes100** [comment]
@rubenskuhl tem informação de alguém operacional na cielo ?

**Comment by rubenskuhl** [comment]
> @rubenskuhl tem informação de alguém operacional na cielo ?

Nosso GC na Cielo não nos passou nem informação de já estar pronto, nem de roadmap ou tarifas. Também não recebi informação da Cielo de outrem, por isso não está na lista... eles tem algo ?

**Comment by rodrigonunes100** [comment]
Eu sou cliente deles, e no site parece que vão liberar o pix, já fiz a solicitação e estou na espera.
https://www.cielo.com.br/pix

**Comment by rodrigonunes100** [comment]
@rubenskuhl você precisa ir em meucadastro / autorizaçoes / PIX   e assinar o  Termo de aceite para as transações via PIX do seu estabelecimento. Eles dão um prazo de 3 dias úteis para resposta. Também não sei se tem API, e se é padrão BACEN. To aguardando...

**Comment by rubenskuhl** [comment]
> @rubenskuhl você precisa ir em meucadastro / autorizaçoes / PIX e assinar o Termo de aceite para as transações via PIX do seu estabelecimento. Eles dão um prazo de 3 dias úteis para resposta. Também não sei se tem API, e se é padrão BACEN. To aguardando...

Quando você tiver uma atualização deles, poderia repassar para eu atualizar as informações do #76 ?

**Comment by rodrigonunes100** [comment]
@rubenskuhl ,ok passo sim.

**Comment by danilolv** [comment]
Hoje, 27/11/2020, foi um marco pra nós aqui da Gerencianet, tivemos o primeiro cliente que completou todo o fluxo homologatório da nossa Api Pix. Ele já está 100% funcional em produção, gerando qrcode dinâmico, recebendo e processando webooks, tudo dentro das especificações do Bacen. O fluxo final ficou bastante robusto.

**Comment by flaviolenz** [comment]
@danilolv , parabens!

Ja abri uma conta com voces e vou fazer a integracao.
Vi que voces enviarão um certificado pra estabelecer o MTLS.
Vai uma sugestao de vcs aceitarem TAMBÉM, se a autenticacao for com um certificado da ICP-Brasil.
Toda empresa tem um certificado desses e as aplicacoes ja o gerenciam, o que diminuiria a necessidade de armazenar outros certificados da empresa, inclusive, um para cada banco.

Tecnicamente falando, seria vc colocar na sua Truststore o raiz da ICP-BR.
E na chamada do webhook, usar o certificado do gerencia net que vcs tambem devem ter.
O cliente colocaria o ICP-BR raiz tb em seu servidor.

**Comment by rubenskuhl** [comment]
> @danilolv , parabens!
> 
> Ja abri uma conta com voces e vou fazer a integracao.
> Vi que voces enviarão um certificado pra estabelecer o MTLS.
> Vai uma sugestao de vcs aceitarem TAMBÉM, se a autenticacao for com um certificado da ICP-Brasil.
> Toda empresa tem um certificado desses e as aplicacoes ja o gerenciam, o que diminuiria a necessidade de armazenar outros certificados da empresa, inclusive, um para cada banco.
> 
> Tecnicamente falando, seria vc colocar na sua Truststore o raiz da ICP-BR.
> E na chamada do webhook, usar o certificado do gerencia net que vcs tambem devem ter.
> O cliente colocaria o ICP-BR raiz tb em seu servidor.

Com o gotcha já mencionado em outro tópico de usar certificado A1 e não A3, que é péssimo para integração server-to-server. 

Mas em sendo como opção e não obrigação para todos os clientes PSP, é questão de ver se o MVP já está consolidado e crescer features.

**Comment by flaviolenz** [comment]
Obvio que eh A1, mas não vem ao caso.
O certificado raiz eh em arquivo.

Ja superamos esse assunto antes.

Em qua, 2 de dez de 2020 12:30, Rubens Kuhl <notifications@github.com>
escreveu:
